From d934a4af404585b4a4543ad670aaa7bd9bedb2cf Mon Sep 17 00:00:00 2001
From: Suman Anna <s-anna@ti.com>
Date: Wed, 11 Nov 2015 16:36:11 -0600
Subject: [PATCH 08/12] HACK: ARM: DRA7: Configure DSP & IVA voltage domains
 for OPP_HIGH

The current bootloaders has all the Voltage domains setup and
configured for OPP_NOM. Update the voltage domain data on DSP
and IVA Voltage domains to be configured for OPP_HIGH. This is
being done to meet the performance needs of 1080p MultiMedia
usecases and other DSP usecases. This change is done for all
the existing DRA7xx/AM57xx platforms.

The OPP_HIGH voltage settings are done in u-boot and the clock
configuration done in kernel because of the following reasons:
1. SoC definition constraints us to NOT to do dynamic voltage
   scaling ever after the initial avs0 setting in bootloader
  - so the voltage must be set in bootloader.
2. The voltage level must be set even if the IP blocks like
   SGX/DSP are unused.
3. The IVA and DSP DPLLs are not essential for u-boot
   functionality, and similar DPLL clock configuration code
   has been cleaned up in v2014.10 kernel. See commit,
   02c4153 ("ARM: OMAP4/5: Remove dead code against
   CONFIG_SYS_CLOCKS_ENABLE_ALL").

The non-essential DPLLs are configured within the kernel during
the clock init step when parsing the device tree and creating
the clock devices. This approach meets both the u-boot and kernel
needs.

The GPU voltage domain is also updated to OPP_HIGH on platforms
with ganged voltage rails supplying either of the IVA or DSP
voltage domains. Following is the complete list of voltage
domains updated:
   AM57xx EVM / Beagle-x15 : IVA, DSPEVE & GPU (SMPS45)
   AM571x / AM572x IDK     : IVA, DSPEVE & GPU (SMPS45)
   DRA72x EVM              : IVA, DSPEVE & GPU (SMPS3)
   DRA74x EVM              : IVA (SMPS8) and DSPEVE (SMPS45)

Signed-off-by: Suman Anna <s-anna@ti.com>
---
 arch/arm/cpu/armv7/omap5/hw_data.c      | 18 +++++++++---------
 arch/arm/include/asm/arch-omap5/clock.h | 10 ++++++++++
 board/ti/am57xx/board.c                 | 12 ++++++------
 3 files changed, 25 insertions(+), 15 deletions(-)

diff --git a/arch/arm/cpu/armv7/omap5/hw_data.c b/arch/arm/cpu/armv7/omap5/hw_data.c
index 7f8c0a4..26a8dc4 100644
--- a/arch/arm/cpu/armv7/omap5/hw_data.c
+++ b/arch/arm/cpu/armv7/omap5/hw_data.c
@@ -369,8 +369,8 @@ struct vcores_data dra752_volts = {
 	.mpu.addr	= TPS659038_REG_ADDR_SMPS12,
 	.mpu.pmic	= &tps659038,
 
-	.eve.value	= VDD_EVE_DRA752,
-	.eve.efuse.reg	= STD_FUSE_OPP_VMIN_DSPEVE_NOM,
+	.eve.value	= VDD_EVE_DRA752_HIGH,
+	.eve.efuse.reg	= STD_FUSE_OPP_VMIN_DSPEVE_HIGH,
 	.eve.efuse.reg_bits	= DRA752_EFUSE_REGBITS,
 	.eve.addr	= TPS659038_REG_ADDR_SMPS45,
 	.eve.pmic	= &tps659038,
@@ -387,8 +387,8 @@ struct vcores_data dra752_volts = {
 	.core.addr	= TPS659038_REG_ADDR_SMPS7,
 	.core.pmic	= &tps659038,
 
-	.iva.value	= VDD_IVA_DRA752,
-	.iva.efuse.reg	= STD_FUSE_OPP_VMIN_IVA_NOM,
+	.iva.value	= VDD_IVA_DRA752_HIGH,
+	.iva.efuse.reg	= STD_FUSE_OPP_VMIN_IVA_HIGH,
 	.iva.efuse.reg_bits	= DRA752_EFUSE_REGBITS,
 	.iva.addr	= TPS659038_REG_ADDR_SMPS8,
 	.iva.pmic	= &tps659038,
@@ -411,20 +411,20 @@ struct vcores_data dra722_volts = {
 	 * The DSPEVE, GPU and IVA rails are usually grouped on DRA72x
 	 * designs and powered by TPS65917 SMPS3, as on the J6Eco EVM.
 	 */
-	.gpu.value	= VDD_GPU_DRA72x,
+	.gpu.value	= VDD_GPU_DRA72x_HIGH,
 	.gpu.efuse.reg	= STD_FUSE_OPP_VMIN_GPU_NOM,
 	.gpu.efuse.reg_bits = DRA752_EFUSE_REGBITS,
 	.gpu.addr	= TPS65917_REG_ADDR_SMPS3,
 	.gpu.pmic	= &tps659038,
 
-	.eve.value	= VDD_EVE_DRA72x,
-	.eve.efuse.reg	= STD_FUSE_OPP_VMIN_DSPEVE_NOM,
+	.eve.value	= VDD_EVE_DRA72x_HIGH,
+	.eve.efuse.reg	= STD_FUSE_OPP_VMIN_DSPEVE_HIGH,
 	.eve.efuse.reg_bits = DRA752_EFUSE_REGBITS,
 	.eve.addr	= TPS65917_REG_ADDR_SMPS3,
 	.eve.pmic	= &tps659038,
 
-	.iva.value	= VDD_IVA_DRA72x,
-	.iva.efuse.reg	= STD_FUSE_OPP_VMIN_IVA_NOM,
+	.iva.value	= VDD_IVA_DRA72x_HIGH,
+	.iva.efuse.reg	= STD_FUSE_OPP_VMIN_IVA_HIGH,
 	.iva.efuse.reg_bits = DRA752_EFUSE_REGBITS,
 	.iva.addr	= TPS65917_REG_ADDR_SMPS3,
 	.iva.pmic	= &tps659038,
diff --git a/arch/arm/include/asm/arch-omap5/clock.h b/arch/arm/include/asm/arch-omap5/clock.h
index 38d50d6..cccbd1b 100644
--- a/arch/arm/include/asm/arch-omap5/clock.h
+++ b/arch/arm/include/asm/arch-omap5/clock.h
@@ -246,6 +246,11 @@
 #define VDD_CORE_DRA752		1060
 #define VDD_IVA_DRA752		1060
 
+/* DRA74x/75x voltage settings in mv for OPP_HIGH per DM */
+#define VDD_EVE_DRA752_HIGH	1250
+#define VDD_GPU_DRA752_HIGH	1250
+#define VDD_IVA_DRA752_HIGH	1250
+
 /* DRA72x voltage settings in mv for OPP_NOM per DM */
 #define VDD_MPU_DRA72x		1100
 #define VDD_EVE_DRA72x		1060
@@ -253,6 +258,11 @@
 #define VDD_CORE_DRA72x		1060
 #define VDD_IVA_DRA72x		1060
 
+/* DRA72x voltage settings in mv for OPP_HIGH per DM */
+#define VDD_EVE_DRA72x_HIGH	1250
+#define VDD_GPU_DRA72x_HIGH	1250
+#define VDD_IVA_DRA72x_HIGH	1250
+
 /* Efuse register offsets for DRA7xx platform */
 #define DRA752_EFUSE_BASE	0x4A002000
 #define DRA752_EFUSE_REGBITS	16
diff --git a/board/ti/am57xx/board.c b/board/ti/am57xx/board.c
index 420203c..4d90020 100644
--- a/board/ti/am57xx/board.c
+++ b/board/ti/am57xx/board.c
@@ -229,14 +229,14 @@ struct vcores_data beagle_x15_volts = {
 	.mpu.addr		= TPS659038_REG_ADDR_SMPS12,
 	.mpu.pmic		= &tps659038,
 
-	.eve.value		= VDD_EVE_DRA752,
-	.eve.efuse.reg		= STD_FUSE_OPP_VMIN_DSPEVE_NOM,
+	.eve.value		= VDD_EVE_DRA752_HIGH,
+	.eve.efuse.reg		= STD_FUSE_OPP_VMIN_DSPEVE_HIGH,
 	.eve.efuse.reg_bits	= DRA752_EFUSE_REGBITS,
 	.eve.addr		= TPS659038_REG_ADDR_SMPS45,
 	.eve.pmic		= &tps659038,
 
-	.gpu.value		= VDD_GPU_DRA752,
-	.gpu.efuse.reg		= STD_FUSE_OPP_VMIN_GPU_NOM,
+	.gpu.value		= VDD_GPU_DRA752_HIGH,
+	.gpu.efuse.reg		= STD_FUSE_OPP_VMIN_GPU_HIGH,
 	.gpu.efuse.reg_bits	= DRA752_EFUSE_REGBITS,
 	.gpu.addr		= TPS659038_REG_ADDR_SMPS45,
 	.gpu.pmic		= &tps659038,
@@ -247,8 +247,8 @@ struct vcores_data beagle_x15_volts = {
 	.core.addr		= TPS659038_REG_ADDR_SMPS6,
 	.core.pmic		= &tps659038,
 
-	.iva.value		= VDD_IVA_DRA752,
-	.iva.efuse.reg		= STD_FUSE_OPP_VMIN_IVA_NOM,
+	.iva.value		= VDD_IVA_DRA752_HIGH,
+	.iva.efuse.reg		= STD_FUSE_OPP_VMIN_IVA_HIGH,
 	.iva.efuse.reg_bits	= DRA752_EFUSE_REGBITS,
 	.iva.addr		= TPS659038_REG_ADDR_SMPS45,
 	.iva.pmic		= &tps659038,
-- 
2.1.4

